integrations/pagerduty/main.py

```diff
@@ -30,16 +30,18 @@ def initialize_client() -> PagerDutyClient:
 async def enrich_service_with_analytics_data(
     client: PagerDutyClient, services: list[dict[str, Any]], months_period: int
 ) -> list[dict[str, Any]]:
-    async def fetch_service_analytics(service: dict[str, Any]) -> dict[str, Any]:
+    async def fetch_service_analytics(services: list[dict[str, Any]]) -> dict[str, Any]:
```

Comment:
let's move the service_ids above the try catch block

`service_ids = [service["id"] for service in services]`